src/execution_plan/ops/op_update.c

```diff
@@ -109,11 +113,36 @@ OpResult OpUpdateReset(OpBase *ctx) {
     return OP_OK;
 }
 
+/* Updates indices for all entities getting modified. This must be invoked
+ * before _UpdateEntities, while we still have access to the original property. */
+void _UpdateIndices(OpUpdate *op) {
+    for(int i = 0; i < op->entities_to_update_count; i++) {
+        // Retrieve and iterate over all labels associated with each node
+        size_t *node_labels = NULL;
+        size_t label_count = Graph_GetNodeLabels(op->g, op->entities_to_update[i].id, &node_labels);
```

Comment:
Preparation for multi label entities?

Reply:
In part, yes, but it's also valid to get to this point without any knowledge of labels with a query like:
`MATCH (a) SET a.counter = a.counter + 1`
Even outside of the context of indexing, actually, label store schemas can become outdated in a similar way:
`MATCH (a) SET a.dup = a.counter`
will cause `dup` to be added to the AllStore, but not any label store. (Comment about this at `op_update.c:166`)